src/module.c

```diff
@@ -5128,7 +5284,7 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
         }
         acl_retval = ACLCheckAllUserCommandPerm(ctx->client->user,c->cmd,c->argv,c->argc,&acl_errpos);
         if (acl_retval != ACL_OK) {
-            sds object = (acl_retval == ACL_DENIED_CMD) ? (sds)c->cmd->name : c->argv[acl_errpos]->ptr;
+            sds object = (acl_retval == ACL_DENIED_CMD) ? sdsnew(c->cmd->name) : sdsdup(c->argv[acl_errpos]->ptr);
```

Comment:
didn't you say that for modules the command name is an sds (hence the casting before this change)?
maybe i'm missing something (superficial review)

Reply:
you are correct but here c->cmd is most likely not a module command (we are in RM_Call)

I could check the CMD_MODULE flag and call sdsnew or sdsdup accordingly...

Reply:
ok, so the reason this casting was working till now is because internally the ACL code didn't use the sds header..
so now it's cleaner, and there's no decrease of efficiency...
thanks.